src/main/java/startsteps/ECommerceShop/config/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,59 @@
+package startsteps.ECommerceShop.config;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.*;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import startsteps.ECommerceShop.service.JwtService;
+import startsteps.ECommerceShop.service.UserService;
+import org.apache.commons.lang3.StringUtils;
+
+
+import java.io.IOException;
+
+@Component
+@AllArgsConstructor
+
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    private final JwtService jwtService;
+    private final UserService userService;
+
+
+    @Override
+    protected void doFilterInternal(@NonNull HttpServletRequest request,
+                                    @NonNull HttpServletResponse response,
+                                    @NonNull FilterChain filterChain)
+            throws ServletException, IOException {
+        final String authHeader = request.getHeader("Authorization");
+        final String jwt;
+        final String userEmail;
+        if (StringUtils.isEmpty(authHeader) || !StringUtils.startsWith(authHeader, "Bearer")) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+        jwt = authHeader.substring(7);
```

Comment:
this can cause an issue issue if the autheader is less than 7 chars.

Reply:
In my case "Bearer " includes 7 characters, and then goes token-genereted code and it is extracts this part to use